publisher/config/schema.go

```diff
@@ -65,3 +73,36 @@ func parseUploadSchema(fileContent []byte) (UploadArtifactSchemas, error) {
 
 	return schema, nil
 }
+
+// ValidateSchemas validates that some components of the schema are correct (this is still wip)
+// It validates:
+//   - no incorrect fileType is present
+//   - the app name is the prefix of the src package. This is mandatory to create consistent apt repositories.
+//     If they are not equal, the file will be uploaded to a location, and the metadata will point to different location
+//     which will break the apt repository as you'll receive a 404 when trying to install the package.
+func ValidateSchemas(appName string, schemas UploadArtifactSchemas) error {
+	for _, schema := range schemas {
+		if !isValidAppName(appName, schema) {
+			return fmt.Errorf("invalid app name: %s", appName)
+		}
+		for _, upload := range schema.Uploads {
+			if !isValidType(upload.Type) {
+				return fmt.Errorf("invalid upload type: %s", upload.Type)
```

Comment:
Same for this error, we could specify what are the valid upload types.

Reply:
I have reviewed the errors, thanks for the feedback :) 3b57e23e12f1f20a1f93a78c229bd49c04dcaa7c